### PR TITLE
feat: Allow users to enable session replay during product analytics onboarding

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

The toggle is gated behind a 50% flag. https://us.posthog.com/project/2/experiments/33719

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by xrdt** [comment]
Not sure why the experiments test keeps failing @neilkakkar @jurajmajerik , am I missing something?

File: frontend/src/scenes/onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -80,39 +80,47 @@ const OnboardingWrapper = ({ children }: { children: React.ReactNode }): JSX.Ele
...
+                   and scrolling to create heatmaps. 
+                   No additional events are created, and you can disable this at any time.`,
+            teamProperty: 'heatmaps_opt_in',
+            value: currentTeam?.heatmaps_opt_in ?? true,
+            type: 'toggle',
+        },
+    ]
+
+    if (featureFlags[FEATURE_FLAGS.ENABLE_SESSION_REPLAY_PA_ONBOARDING]) {
+        options.push({
```
Comment (xrdt) [inline frontend/src/scenes/onboarding/Onboarding.tsx:108]: Should we also enable console logs, network performance?

Comment (pauldambra) [reply frontend/src/scenes/onboarding/Onboarding.tsx:108]: yep iirc the default in replay is console logs and "performance" opt in, not headers or body capture

**Review by pauldambra — APPROVED** [review]
didn't run it but looks good to me

File: frontend/src/scenes/onboarding/Onboarding.tsx (hunk 1 above)
Comment (xrdt) [reply frontend/src/scenes/onboarding/Onboarding.tsx:110]: Ok, added.

File: frontend/src/scenes/onboarding/OnboardingProductConfiguration.tsx [hunk 2]
```diff
@@ -62,21 +62,23 @@ export const OnboardingProductConfiguration = ({
...
+                value: option.value,
+                onChange: (newValue: boolean | string | number) => {
+                    // Use the current value from the ref to ensure that onChange always accesses
+                    // the latest state of configOptions, preventing the closure from using stale data.
+                    const updatedConfigOptions = configOptionsRef.current.map((o) =>
+                        o.teamProperty === option.teamProperty ? { ...o, value: newValue } : o
+                    )
+                    setConfigOptions(updatedConfigOptions)
+                },
+            })),
```
Comment (raquelmsmith) [inline frontend/src/scenes/onboarding/OnboardingProductConfiguration.tsx:81]: @zlwaterfield when we move to a single subscribe, we should make sure people are aware of the potential costs here when enabling replay from analytics onboarding.

**Pull request merged**